### ISSUE TITLE
`onChange` behavior

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#1942]
### What version of Hono are you using?

4.0.0-rc.0

### What runtime/platform is your app running on?

any

### What steps can reproduce the bug?

This is not a bug, but a request.

With the following code, the behavior of the `onChange` event is different from React.

```tsx
import { render } from 'hono/jsx/dom'
import { useState } from 'hono/jsx/hooks'

const root = document.getElementById('root') as HTMLElement

const TextArea = () => {
  const [value, setValue] = useState('')

  const handleChange = (e: Event) => {
    const target = e.target as HTMLTextAreaElement
    setValue(target.value)
  }

  return (
    <div>
      <textarea onChange={handleChange} />
      <div>{value}</div>
    </div>
  )
}

render(<TextArea />, root)
```

### What is the expected behavior?

This is the React code and the demo.

https://github.com/honojs/hono/assets/10682/bfb370fa-2c76-4444-9196-d49208ee7809

```tsx
import React, { useState } from 'react'
import { hydrateRoot } from 'react-dom/client'

const root = document.getElementById('root') as HTMLElement

const TextArea = () => {
  const [value, setValue] = useState('')

  const handleChange = (e: Event) => {
    const target = e.target as HTMLTextAreaElement
    setValue(target.value)
  }

  return (
    <div>
      <textarea onChange={handleChange} />
      <div>{value}</div>
    </div>
  )
}

hydrateRoot(root, <TextArea />)
```

### What do you see instead?

This is the `hono/jsx/demo` demo.  I need to click outside of the `textarea` to trigger the `onChange` event:

https://github.com/honojs/hono/assets/10682/b24323be-77ef-4f3d-82b6-1423dc595561

### Additional information

This may be a specification issue, but since I want to get a value for each input, I think it should match React's behavior.

**Comment by usualoma** [comment]
👀

**Comment by yusukebe** [comment]
@usualoma 

What do you think of this?

**Comment by yusukebe** [comment]
Fixed! But, another issue came up: #2016
